internal/saucecloud/cypress.go

```diff
@@ -23,39 +22,38 @@ func (r *CypressRunner) RunProject() (int, error) {
 	var deprecationMessage string
 	exitCode := 1
 
-	m, err := r.MetadataSearchStrategy.Find(context.Background(), r.MetadataService, cypress.Kind, r.Project.Cypress.Version)
+	cyVersion := r.Project.GetVersion()
+	m, err := r.MetadataSearchStrategy.Find(context.Background(), r.MetadataService, cypress.Kind, cyVersion)
 	if err != nil {
 		r.logFrameworkError(err)
 		return exitCode, err
 	}
-	r.Project.Cypress.Version = m.FrameworkVersion
-	if r.Project.RunnerVersion == "" {
-		r.Project.RunnerVersion = m.CloudRunnerVersion
+	r.Project.SetVersion(m.FrameworkVersion)
+	if r.Project.GetRunnerVersion() == "" {
+		r.Project.SetRunnerVersion(m.CloudRunnerVersion)
 	}
 
 	if m.Deprecated {
-		deprecationMessage = r.deprecationMessage(cypress.Kind, r.Project.Cypress.Version)
+		deprecationMessage = r.deprecationMessage(cypress.Kind, cyVersion)
 		fmt.Print(deprecationMessage)
 	}
 
-	for _, s := range r.Project.Suites {
-		if s.PlatformName != "" && !framework.HasPlatform(m, s.PlatformName) {
-			msg.LogUnsupportedPlatform(s.PlatformName, framework.PlatformNames(m.Platforms))
-			return 1, errors.New("unsupported platform")
+	if err := r.validateTunnel(r.Project.GetSauceCfg().Tunnel.Name, r.Project.GetSauceCfg().Tunnel.Owner); err != nil {
+		for _, s := range r.Project.GetSuites() {
+			if s.PlatformName != "" && !framework.HasPlatform(m, s.PlatformName) {
+				msg.LogUnsupportedPlatform(s.PlatformName, framework.PlatformNames(m.Platforms))
+				return 1, errors.New("unsupported platform")
+			}
```

Comment:
This block is wrong ⚠️ Not sure why the platform validation is inside the conditional error-block of the tunnel validation?

Reply:
resolving